docs/RationaleSymbols.md

```diff
@@ -35,32 +35,49 @@ symbol-like data to allow forming non-SSA linkage between disparate elements.
 To accomplish this, an attribute named `inner_sym` is attached, providing a 
 scoped symbol-like name to the element.  An operation with an `inner_sym`
 resides in arbitrarily-nested regions of a region that defines an
-`InnerSymbolTable` and a `Symbol` .  `InnerSymbolTable` operations must reside
-within an `InnerRefNamespace`.  The `inner_sym` attribute must be an `InnerSymAttr`
-which defines the inner symbols attached to the operation and its fields.
-Operations containing an inner symbol must implement the `InnerSymbol` interface.
+`InnerSymbolTable`.  `InnerSymbolTable` operations can themselves be arbitrarily
+nested, and thus must reside within an `InnerRefNamespace` or `InnerSymbolTable`.
+The `inner_sym` attribute must be an `InnerSymAttr` which defines the inner
+symbols attached to the operation and its fields. Operations containing an inner
+symbol must implement the `InnerSymbol` interface.
 
 Inner Symbols are different from normal symbols due to MLIR symbol table 
 resolution rules.  Specifically, normal symbols are resolved by first going up 
 to the closest parent symbol table and resolving down from there (recursing 
-back down for nested symbol paths).  In FIRRTL and HW, modules define a symbol in a 
-`firrtl.circuit` or `builtin.module` symbol table.  For instances to be able to resolve the 
+back down for nested symbol paths). Conversely, Inner symbol symbol resolution
+is rooted at some encapsulating `InnerRefNamespace`, which defines the top-level
+symbol table hierarchy. In FIRRTL and HW, modules define a symbol in a
+`firrtl.circuit` or `builtin.module` symbol table, which is then interpreted as an
+`InnerRefNamespace`.  For instances to be able to resolve the 
 modules they instantiate, the symbol use in an instance must resolve in the 
-top-level symbol table.  If a module were a symbol table, instances resolving a 
+top-level `InnerRefNamespace`.  If a module were a symbol table, instances resolving a 
```

Comment:
They are, right now, real symbols and resolve using symbol resolution in symbol tables.